api/client/node/next.ts

```diff
@@ -0,0 +1,77 @@
+import type { OpenAIStreamCallbacks } from 'ai'
+import type { TiktokenModel } from 'js-tiktoken'
+
+let encodingForModel: (model: TiktokenModel) => any | undefined
+
+type OpenAIUsage = {
+  total_tokens: number
+  prompt_tokens: number
+  completion_tokens: number
+}
+
+type OpenAIStreamCallbacksWithUsage = OpenAIStreamCallbacks & {
+  onUsage?: (usage: OpenAIUsage) => void
+}
+
+export function createOpenAIStreamCallback(
+  {
+    model,
+    prompts,
+  }: {
+    model: TiktokenModel
+    prompts: string[]
+  },
+  openAIStreamCallbacks: OpenAIStreamCallbacksWithUsage
+) {
+  // Tiktoken is an optional dependency, so we require it conditionally
+  if (!encodingForModel) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { encodingForModel: encodingForModel_ } = require('js-tiktoken')
+    encodingForModel = encodingForModel_
+  }
+
+  const enc = encodingForModel(model)
+  let promptTokens = 0
+  let completionTokens = 0
+
+  const streamCallbacks: OpenAIStreamCallbacks = {
+    ...openAIStreamCallbacks,
+
+    onStart() {
+      for (const content of prompts) {
+        const tokens = enc.encode(content)
+        promptTokens += tokens.length
+      }
+
+      if (typeof openAIStreamCallbacks?.onStart === 'function') {
+        return openAIStreamCallbacks.onStart()
+      }
+    },
+    onToken: (content) => {
```

Comment:
To keep it consistent
```suggestion
    onToken (content) {
```